jgrapht-core/src/test/java/org/jgrapht/generate/KleinbergSmallWorldGraphGeneratorTest.java

```diff
@@ -0,0 +1,218 @@
+            { 10, 6 }, { 10, 8 }, { 10, 9 }, { 10, 11 }, { 10, 12 }, { 10, 14 }, { 10, 15 },
+            { 10, 16 }, { 10, 20 }, { 11, 1 }, { 11, 5 }, { 11, 6 }, { 11, 7 }, { 11, 9 },
+            { 11, 10 }, { 11, 12 }, { 11, 13 }, { 11, 15 }, { 11, 16 }, { 11, 17 }, { 11, 21 },
+            { 12, 2 }, { 12, 6 }, { 12, 7 }, { 12, 8 }, { 12, 10 }, { 12, 11 }, { 12, 13 },
+            { 12, 14 }, { 12, 16 }, { 12, 17 }, { 12, 18 }, { 12, 22 }, { 13, 3 }, { 13, 7 },
+            { 13, 8 }, { 13, 9 }, { 13, 11 }, { 13, 12 }, { 13, 14 }, { 13, 15 }, { 13, 17 },
+            { 13, 18 }, { 13, 19 }, { 13, 23 }, { 14, 4 }, { 14, 8 }, { 14, 9 }, { 14, 10 },
+            { 14, 12 }, { 14, 13 }, { 14, 15 }, { 14, 16 }, { 14, 18 }, { 14, 19 }, { 14, 20 },
+            { 14, 24 }, { 15, 5 }, { 15, 9 }, { 15, 10 }, { 15, 11 }, { 15, 13 }, { 15, 14 },
+            { 15, 16 }, { 15, 17 }, { 15, 19 }, { 15, 20 }, { 15, 21 }, { 16, 6 }, { 16, 10 },
+            { 16, 11 }, { 16, 12 }, { 16, 14 }, { 16, 15 }, { 16, 17 }, { 16, 18 }, { 16, 20 },
+            { 16, 21 }, { 16, 22 }, { 17, 7 }, { 17, 11 }, { 17, 12 }, { 17, 13 }, { 17, 15 },
+            { 17, 16 }, { 17, 18 }, { 17, 19 }, { 17, 21 }, { 17, 22 }, { 17, 23 }, { 18, 8 },
+            { 18, 12 }, { 18, 13 }, { 18, 14 }, { 18, 16 }, { 18, 17 }, { 18, 19 }, { 18, 20 },
+            { 18, 22 }, { 18, 23 }, { 18, 24 }, { 19, 9 }, { 19, 13 }, { 19, 14 }, { 19, 15 },
+            { 19, 17 }, { 19, 18 }, { 19, 20 }, { 19, 21 }, { 19, 23 }, { 19, 24 }, { 20, 10 },
+            { 20, 14 }, { 20, 15 }, { 20, 16 }, { 20, 18 }, { 20, 19 }, { 20, 21 }, { 20, 22 },
+            { 20, 24 }, { 21, 11 }, { 21, 15 }, { 21, 16 }, { 21, 17 }, { 21, 19 }, { 21, 20 },
+            { 21, 22 }, { 21, 23 }, { 22, 12 }, { 22, 16 }, { 22, 17 }, { 22, 18 }, { 22, 20 },
+            { 22, 21 }, { 22, 23 }, { 22, 24 }, { 23, 13 }, { 23, 17 }, { 23, 18 }, { 23, 19 },
+            { 23, 21 }, { 23, 22 }, { 23, 24 }, { 24, 14 }, { 24, 18 }, { 24, 19 }, { 24, 20 },
+            { 24, 22 }, { 24, 23 }, { 0, 12 }, { 2, 5 }, { 3, 23 }, { 3, 22 }, { 4, 22 }, { 5, 22 },
+            { 8, 11 }, { 13, 16 }, { 14, 2 }, { 15, 12 }, { 21, 7 }, { 22, 19 }, { 23, 2 } };
+
+        assertEquals(25, g.vertexSet().size());
+        for (int[] e : edges) {
+            assertTrue(g.containsEdge(e[0], e[1]));
+        }
+        assertEquals(edges.length, g.edgeSet().size());
+
+    }
+
+    @Test
+    public void testDirectedWithUniform()
+    {
+        final long seed = 5;
+
+        GraphGenerator<Integer, DefaultEdge, Integer> gen =
+            new KleinbergSmallWorldGraphGenerator<>(5, 2, 3, 0, seed);
+        Graph<Integer, DefaultEdge> g = new SimpleDirectedGraph<>(DefaultEdge.class);
+        gen.generateGraph(g, new IntegerVertexFactory(), null);
+
+        int[][] edges = { { 0, 1 }, { 0, 2 }, { 0, 4 }, { 0, 5 }, { 0, 6 }, { 0, 10 }, { 1, 0 },
```

Comment:
Did you use some external method to generate this result?
Obviously, the following approach wouldn't be good for testing: 1. implement a function `f(x)`, 2. use this function to compute a result f(y)=z, 3. add a test method which verifies that f(x) is equal to z. If 'z' was the wrong solution in the first place, then this test would simply ensure that the method continues to produce the wrong result.

I realize it may be hard to come up with tests for these generators, but making errors in some of the many parameters/for loop indices is quite easy and very hard to spot while reviewing.